web3/middleware/filter.py

```diff
@@ -0,0 +1,324 @@
+
+    starting from `fromBlock` to the latest mined block,
+    until reaching toBlock. e.g.:
+
+
+    >>> blocks_to_filter = unload_latest_blocks(w3, 0, 50)
+    >>> next(blocks_to_filter)  # latest block number = 11
+    (0, 11)
+    >>> next(blocks_to_filter)  # latest block number = 45
+    (12, 45)
+    >>> next(blocks_to_filter)  # latest block number = 50
+    (46, 50)
+    """
+    _from_block = from_block
+    for _latest_block in iter_latest_block(web3, to_block):
+        if _latest_block is None:
+            yield (None, None)
+        else:
+            yield (_from_block, _latest_block)
+            _from_block = _latest_block + 1
+
+
+def get_logs_multipart(
+        web3,
+        startBlock,
+        stopBlock,
+        address,
+        topics,
+        max_blocks):
+    """Use to make large requests to ``eth_getLogs``
+
+    The getLog request is broken into multiple calls of the max number of blocks
+    ``max_blocks``.
+    """
+    chunks = block_range_chunker(startBlock, stopBlock, max_blocks)
+    for from_block, to_block in chunks:
+        params = {
+            'fromBlock': from_block,
+            'toBlock': to_block,
+            'address': address,
+            'topics': topics
+        }
+        yield web3.eth.getLogs(params)
+
+
+class RequestLogs:
+    def __init__(
+            self,
+            web3,
+            from_block=None,
+            to_block=None,
+            address=None,
+            topics=None):
+
+        self.address = address
+        self.topics = topics
+        self.web3 = web3
+        if from_block is None or from_block == 'latest':
+            self._from_block = web3.eth.blockNumber
+        else:
+            self._from_block = from_block
+        self._to_block = to_block
+        self.filter_changes = self._get_filter_changes()
+
+    @property
+    def from_block(self):
+        if self._from_block > self.web3.eth.blockNumber:
+            from_block = self.web3.eth.blockNumber
+
+        else:
+            from_block = self._from_block
+
+        return from_block
+
+    @property
+    def to_block(self):
+        if self._to_block is None:
+            to_block = self.web3.eth.blockNumber
+
+        elif self._to_block == 'latest':
+            to_block = self.web3.eth.blockNumber
+
+        elif self._to_block > self.web3.eth.blockNumber:
+            to_block = self.web3.eth.blockNumber
+
+        else:
+            to_block = self._to_block
+
+        return to_block
+
+    def _get_filter_changes(self):
+        for start, stop in iter_latest_block_ranges(self.web3, self.from_block, self._to_block):
+            if None in (start, stop):
+                yield []
+
+            yield list(
+                concat(
+                    get_logs_multipart(
+                        self.web3,
+                        start,
+                        stop,
+                        self.address,
+                        self.topics,
+                        max_blocks=5)))
```

Comment:
This value should probably be configurable.  Think about requesting logs over the span of 10,000 blocks, that's 2,000 API requests.  I'd at least up the default to around 100.